src/overlays/ovl_i4/fox_ka.c

```diff
@@ -1,63 +1,2183 @@
+
+        case 3:
+            state = 1;
+            if (actor->timer_0BC == 0) {
+                actor->fwork[3] = 1.2f;
+                actor->fwork[1] = 40.0f;
+                yRand = RAND_FLOAT(1000.0f);
+                if (gPlayer[0].state_1C8 == PLAYERSTATE_1C8_0) {
+                    xRand = RAND_FLOAT_CENTERED(5000.0f);
+                    zRand = RAND_FLOAT_CENTERED(5000.0f);
+                } else {
+                    xRand = RAND_FLOAT_CENTERED(10000.0f);
+                    zRand = RAND_FLOAT_CENTERED(10000.0f);
+                }
+                actor->fwork[4] = xRand;
+                actor->fwork[5] = yRand;
+                actor->fwork[6] = zRand;
+                actor->timer_0BC = RAND_INT(20.0f) + 10;
+            }
+
+            if (actor->timer_0C0 == 0) {
+                actor->timer_0C0 = RAND_INT(200.0f) + 200;
+                actor->fwork[10] = 30.0f;
+            }
+
+            if ((actor->unk_0E6 > 0) && (gActors[actor->unk_0E6].obj.status == 2)) {
+                actor->state = 2;
+                actor->iwork[2] = 0;
+            }
+            break;
+    }
+
+    xSin = __sinf(actor->obj.rot.x * M_DTOR);
+    xCos = __cosf(actor->obj.rot.x * M_DTOR);
+    ySin = __sinf(actor->obj.rot.y * M_DTOR);
+    yCos = __cosf(actor->obj.rot.y * M_DTOR);
+
+    if (state != 0) {
+        xRand = actor->fwork[4] - actor->obj.pos.x;
+        yRand = actor->fwork[5] - actor->obj.pos.y;
+        zRand = actor->fwork[6] - actor->obj.pos.z;
+        if (!((actor->index + gGameFrameCount) & 7)) {
+            actor->fwork[19] = Math_RadToDeg(Math_Atan2F(xRand, zRand));
+            xAngle = sqrtf(SQ(xRand) + SQ(zRand));
+            actor->fwork[20] = Math_RadToDeg(Math_Atan2F(yRand, xAngle));
+        }
+        xAngle = actor->fwork[20];
+        isClose = func_i4_801989F4(actor, ySin, yCos);
+        if (isClose != 0) {
+            xAngle += 40.0f * isClose;
+            if (xAngle >= 360.0f) {
+                xAngle -= 360.0f;
+            }
+
+            if (xAngle < 0.0f) {
+                xAngle += 360.0f;
+            }
+        } else if ((actor->obj.pos.y < (gGroundLevel + 50.0f)) && (xAngle > 180.0f)) {
+            xAngle = 0.0f;
+            actor->unk_0F4.x = 0.0f;
+        }
+        Math_SmoothStepToAngle(&actor->unk_0F4.x, xAngle, 0.5f, actor->fwork[2], 0.0001f);
+        yAngle = Math_SmoothStepToAngle(&actor->unk_0F4.y, actor->fwork[19], 0.5f, actor->fwork[2], 0.0001f) * 30.0f;
+        if (yAngle < 0.0f) {
+            zAngle = yAngle * -1.0f;
+        } else {
+            zAngle = 360.0f - yAngle;
+        }
+        Math_SmoothStepToAngle(&actor->obj.rot.z, zAngle, 0.1f, 3.0f, 0.01f);
+    }
+    actor->obj.rot.x = -actor->unk_0F4.x;
+    actor->obj.rot.y = actor->unk_0F4.y;
+
+    Math_SmoothStepToF(&actor->fwork[0], actor->fwork[1], 0.2f, 1.0f, 0.1f);
+    Math_SmoothStepToF(&actor->fwork[2], actor->fwork[3], 1.0f, 0.1f, 0.1f);
+    zVel = (actor->fwork[0] + actor->fwork[9]) * xCos;
```

Comment:
I would comment here about this being xzVel, but I know the variable reuse is required to match. ah, devs.